### PR TITLE
feat(workflows): auto-release

### PR DESCRIPTION
**Pull request by SunsetTechuila**

### CODE REVIEW

File: .github/workflows/auto_release.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        with:
+          fetch-tags: true
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+      - name: Check the package version
+        id: check_result
+        run: |
+          latest_release_version=$(git for-each-ref --sort=-creatordate --count 1 --format="%(refname:short)" "refs/tags/*")
```
Comment (theRealPadster) [inline .github/workflows/auto_release.yml:27]: Does this require us to push a new tag as well? I always just do the "generate a new tag on release publish" option

Comment (SunsetTechuila) [reply .github/workflows/auto_release.yml:27]: No, you just create a version bump commit and that's it

Comment (theRealPadster) [reply .github/workflows/auto_release.yml:27]: Ah yes, this is getting the latest current tag, I see it. I'm tired 🙃

**Review by theRealPadster — APPROVED** [review]

File: .github/workflows/auto_release.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      contents: write
+    steps:
+      - name: Wait for "Push" workflow
+        uses: lewagon/wait-on-check-action@v1.3.1
+        with:
+          ref: ${{ github.ref }}
+          check-name: Push
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          wait-interval: 10
+      - name: Download "dist" branch
```
Comment (rxri) [inline .github/workflows/auto_release.yml:51]:
```suggestion
      - name: Download target branch

```

**Review by rxri — APPROVED** [review]

**Pull request merged**